_benchmark/workloads/vectorsearch.md

```diff
@@ -0,0 +1,572 @@
+| `query_body`                              | The JSON properties that will be merged with the search body.                                                                                        |
+| `search_clients`                          | The number of clients used to run queries.                                                                                               |
+| `repetitions`                             | The number of repetitions completed until the dataset is exhausted. Default is `1`.                                                                      |
+| `target_throughput`                       | The target throughput for each query operation, in requests per second. Default is `10`.                                                         |
+| `time_period`                             | The period of time dedicated to running the benchmark test, in seconds. Default is `900`.                                                       |
+
+
+
+## Test procedures
+
+The workload supports the following test procedures.
+
+### No train test
+
+The no train test procedure tests vector search indexes that require no training. You can define the underlying configuration of the vector search algorithm (such as specifying a specific engine or space type) as method definitions. 
+
+### No train test (index only)
+
+The no train test (index only) procedure is used to index vector search indexes that require no training. This can be particularly useful when you want to benchmark only the indexing operation.
+
+### No train test (AOSS)
+
+The no train test (AOSS) procedure is similar to the no train test procedure but is specifically used for Amazon OpenSearch Serverless vector search collections. This procedure doesn't include operations like **refresh** and **warmup** because they aren't supported by vector search collections.
+
+### Force merge index
+
+The Force Merge Index procedure optimizes vector search indexes by performing force merge operations up to a given maximum number of segments. For large datasets, force merging is a costly operation. Therefore, we recommend using a separate procedure to occasionally trigger force merge operations based on user requirements.
+
+### Train test
+
+The Train Test procedure benchmarks approximate k-NN search algorithms that require a training step. For example, the Faiss IVF requires a training step to cluster vectors. After the step is performed, the benchmark can search against a smaller number of cluster centroids instead of the entire dataset.
+
+### Search
+
+The Search procedure benchmarks previously indexed vector search indexes. This can be useful when you want to benchmark large vector search indexes without reindexing each time because load time can be substantial for large datasets. This procedure includes warmup operations intended to avoid cold start problems during vector search.
```

Comment:
```suggestion
The search procedure benchmarks previously indexed vector search indexes. This can be useful when you want to benchmark large vector search indexes without reindexing each time because load time can be substantial for large datasets. This procedure includes warmup operations intended to avoid cold start problems during vector search.
```